distributed/scheduler.py

```diff
@@ -4563,6 +4815,15 @@ async def add_nanny(self, comm):
         }
         return msg
 
+    def get_task_duration(self, ts: TaskState) -> double:
+        return self.state.get_task_duration(ts)
+
+    def get_comm_cost(self, *args, **kwargs):
+        return self.state.get_comm_cost(*args, **kwargs)
+
+    def check_idle_saturated(self, *args, **kwargs):
+        return self.state.check_idle_saturated(*args, **kwargs)
+
     def update_graph_hlg(
```

Comment:
The entire `update_graph_*` functions appear to be much close to state than server. Why not move these to `SchedulerState`?